components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/FirefoxAccount.kt

```diff
@@ -3,117 +3,94 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 package mozilla.components.service.fxa
+import kotlinx.coroutines.experimental.Deferred
+import kotlinx.coroutines.experimental.async
+import org.mozilla.fxaclient.internal.FirefoxAccount as InternalFxAcct
 
 /**
  * FirefoxAccount represents the authentication state of a client.
- *
- * @param <T> The type of the value delivered via the FxaResult.
  */
-class FirefoxAccount(override var rawPointer: RawFxAccount?) : RustObject<RawFxAccount>() {
+class FirefoxAccount internal constructor(private val inner: InternalFxAcct): AutoCloseable {
```

Comment:
We'd want some test coverage for this type, but it'll probably make more sense to add tests to the higher-level API when we build it.

Reply:
Test coverage is a bit tricky right now due to https://github.com/mozilla/application-services/issues/262

We could run tests in e.g. androidTest, but not as part of unit tests.